src/cmd/services/m3coordinator/ingest/carbon/ingest.go

```diff
@@ -232,7 +337,11 @@ func (i *ingester) write(
 		}
 	}()
 
-	for _, rule := range i.rules {
+	i.RLock()
```

Comment:
is there any race now where a connection is handled before the rules are set?

Reply:
No. If present, rules are generated on ingester construction.